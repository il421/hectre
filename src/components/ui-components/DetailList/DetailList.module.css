@value variables: "../../../common/variables.css";
@value xs-size, l-size, natural, font-size-xxsmall from variables;

.detailList {
    display: flex;
    flex-direction: column;

    padding-left: l-size;
    padding-right: l-size;

    font-size: font-size-xxsmall;
}

.detailListHeader {
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    text-transform: uppercase;
    font-weight: bold;
}

.detailListBody{
    display: flex;
    flex-direction: column;
}

.detailListRow{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.detailListRow:nth-child(odd) {
    background-color: natural;
}

.detailListItem {
    padding: xs-size;
}
