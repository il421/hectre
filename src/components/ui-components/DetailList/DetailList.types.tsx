export type Column = {
  name: string;
  minWidth: number;
  maxWidth: number;
  onRender: (item: any) => JSX.Element | string;
};

export interface DetailListProps {
  items: object[];
  columns: Column[];
}
