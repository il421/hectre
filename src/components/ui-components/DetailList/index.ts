export * from "./DetailList.module.css";
export * from "./DetailList";
export * from "./DetailList.types";
