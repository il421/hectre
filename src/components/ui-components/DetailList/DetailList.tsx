import { FunctionComponent } from "react";

import styles from "./DetailList.module.css";
import { DetailListProps } from "./DetailList.types";

export const DetailList: FunctionComponent<DetailListProps> = ({
  columns,
  items
}) => {
  return (
    <div className={styles.detailList}>
      {/* header */}
      <div className={styles.detailListHeader}>
        {columns.map((col, idx) => (
          <div
            style={{ minWidth: col.minWidth, maxWidth: col.maxWidth }}
            className={styles.detailListItem}
            key={`head-${col.name + idx}`}
          >
            {col.name}
          </div>
        ))}
      </div>

      {/* rows */}
      <div className={styles.detailListBody}>
        {items.map((item, idx) => (
          <div key={`row-${idx}`} className={styles.detailListRow}>
            {columns.map((col, idx) => (
              <div
                style={{ minWidth: col.minWidth, maxWidth: col.maxWidth }}
                className={styles.detailListItem}
                key={`row-col-${idx}`}
              >
                {col.onRender(item)}
              </div>
            ))}
          </div>
        ))}
      </div>
    </div>
  );
};
