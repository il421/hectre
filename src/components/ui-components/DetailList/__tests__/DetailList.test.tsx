import { shallow, ShallowWrapper } from "enzyme";

import { DetailList } from "../DetailList";

test("should match previous snapshot", () => {
  const wrapper: ShallowWrapper<typeof DetailList> = shallow<typeof DetailList>(
    <DetailList
      items={[
        { text1: "text1", text2: "text2" },
        { text1: "text3", text2: "text4" }
      ]}
      columns={[
        {
          name: "title1",
          maxWidth: 0,
          minWidth: 0,
          onRender: item => item.text1
        },
        {
          name: "title2",
          maxWidth: 0,
          minWidth: 0,
          onRender: item => item.text2
        }
      ]}
    />
  );
  expect(wrapper.debug()).toMatchSnapshot();
});
