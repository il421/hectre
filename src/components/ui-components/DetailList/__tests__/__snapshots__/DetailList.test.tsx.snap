// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
"<div className=\\"detailList\\">
  <div className=\\"detailListHeader\\">
    <div style={{...}} className=\\"detailListItem\\">
      title1
    </div>
    <div style={{...}} className=\\"detailListItem\\">
      title2
    </div>
  </div>
  <div className=\\"detailListBody\\">
    <div className=\\"detailListRow\\">
      <div style={{...}} className=\\"detailListItem\\">
        text1
      </div>
      <div style={{...}} className=\\"detailListItem\\">
        text2
      </div>
    </div>
    <div className=\\"detailListRow\\">
      <div style={{...}} className=\\"detailListItem\\">
        text3
      </div>
      <div style={{...}} className=\\"detailListItem\\">
        text4
      </div>
    </div>
  </div>
</div>"
`;
